apps/server/src/shared/domain/entity/lesson.entity.ts

```diff
@@ -137,6 +143,14 @@ export class Lesson extends BaseEntityWithTimestamps implements ILearnroomElemen
 		return tasks;
 	}
 
+	getLessonMaterials(): Material[] {
+		if (!this.materials.isInitialized(true)) {
+			throw new Error('Lessons trying to access their materials that are not loaded.');
```

Comment:
we should use more specific errors, in this case a 500
`import { InternalServerErrorException } from '@nextjs/common';`
`throw new InternalServerErrorException('Lessons trying to access their materials that are not loaded.')`

Reply:
should `getTasksItems` also be changed to that error?